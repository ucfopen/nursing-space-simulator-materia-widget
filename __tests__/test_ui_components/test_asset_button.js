import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import AssetButton from "../../src/js/components/ui/asset_button";

// Constants used across all tests
const item = {
	buttonSource: "/images/test.jpg",
	category: "beds",
	id: "bed-1",
	title: "Bed-1"
};

let clickEventMock = jest.fn();


describe("Asset Button Tests", () => {
	test("renders asset button", () => {
		const tree = renderer
			.create(<AssetButton onClick={clickEventMock} item={item} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test("handles click", () => {
		const assetButton = shallow(
			<AssetButton onClick={clickEventMock} item={item} />
		);

		assetButton.find("button").simulate("click");

		expect(clickEventMock).toBeCalled();
	});
});
