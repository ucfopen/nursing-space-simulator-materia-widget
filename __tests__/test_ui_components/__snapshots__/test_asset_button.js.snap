// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Button Tests renders asset button 1`] = `
<div
  className="button-container"
>
  <button
    className="asset"
    data-category="beds"
    data-title="Bed-1"
    id="bed-1"
    onClick={[Function]}
    style={
      Object {
        "background": "url(/images/test.jpg) no-repeat center center",
        "backgroundSize": "100% 100%",
      }
    }
  />
  <div>
    Bed-1
  </div>
</div>
`;
